resample/jackknife.py

```diff
@@ -1,31 +1,50 @@
-from typing import Callable
+"""
+Jackknife resampling.
+"""
 
+from typing import Callable, Sequence
 import numba as nb
 import numpy as np
 
 
-@nb.njit
-def _jackknife_generator(a: np.ndarray) -> np.ndarray:
-    n = len(a)
-    x = np.empty(n - 1, dtype=a.dtype)
-    for i in range(n - 1):
-        x[i] = a[1 + i]
-    # yield x0
-    yield x.view(x.dtype)  # must return view to avoid a numba life-time bug
+def resample(sample: Sequence) -> np.ndarray:
+    """
+    Generator of jackknife'd samples.
 
-    # update of x needs to change values only up to i
-    # for a = [0, 1, 2, 3]
-    # x0 = [1, 2, 3] (yielded above)
-    # x1 = [0, 2, 3]
-    # x2 = [0, 1, 3]
-    # x3 = [0, 1, 2]
-    for i in range(1, n):
-        for j in range(i):
-            x[j] = a[j]
-        yield x.view(x.dtype)  # must return view to avoid a numba life-time bug
+    Parameters
+    ----------
+    sample: array-like
+        Sample. If the sequence is multi-dimensional, the first dimension must
+        walk over i.i.d. observations.
```

Comment:
Being a bit pedantic, but would it be more correct to simply call these "the individual observations" or something like that rather than "i.i.d. observations" since we don't in general know the statistical properties of the data (maybe they're in fact dependent)?

Reply:
If the bootstrap/jackknife are to work correctly, the observations must be i.i.d., right? (ignoring stratification for a moment, which in the end is just a way to fix that stratified data is not i.i.d., but consists of classes of i.i.d. observations)